### DISCUSSION TITLE
Using typescript for template args

### DISCUSSION THREAD

**Discussion by jimmy-e** [storybookjs/storybook#13931] [Help]
Hi all! How can I leverage typescript to accurately detect what my `args` should be?

E.g. if my interface is:
```js
interface Question {
  fieldType: string;
  question: string;
  id: string;
}
```

in the below component, I'm currently not getting any errors stating that I'm missing the `id` field for the `DefaultQuestionCard` template. Thanks!

```js
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Meta } from '@storybook/react/types-6-0';
import QuestionCard from '../QuestionCard';
import { Question } from '../../types';

export default {
  title: 'Question Cards',
} as Meta;

const Template: React.FC<Question> = (props) => <QuestionCard {...props} />;

export const DefaultQuestionCard = Template.bind({});
DefaultQuestionCard.args = {
  fieldType: 'text',
  question: 'Lorem ipsum dolor sit amet?',
};
```

**Comment by shilman** [comment]
Try:

```tsx
import { Meta, Story } from '@storybook/react/types-6-0';
...
const Template: Story<Question> = (props) => <QuestionCard {...props} />;

export const DefaultQuestionCard = Template.bind({});
DefaultQuestionCard.args = {
  fieldType: 'text',
  question: 'Lorem ipsum dolor sit amet?',
};
```

**Reply by jimmy-e** [reply]
unfortunately I'm not seeing any difference.

**Reply by anki247** [reply]
@shilman  `DefaultQuestionCard.args` is still type of `any`! So there is no type support when using  **Args** ...?!
Or do I miss something? 😕

**Reply by shilman** [reply]
If the args annotation is not typed by `Story<Question>` then you need to run typescript in strict mode

https://discord.com/channels/486522875931656193/490070912448724992/773894059588583444
